builtin/common/settings/components.lua

```diff
@@ -442,6 +442,51 @@ local function make_noise_params(setting)
 	}
 end
 
+function make.key(setting)
+	local btn_bind = "bind_" .. setting.name
+	local btn_clear = "unbind_" .. setting.name
+	return {
+		info_text = setting.comment,
+		setting = setting,
+
+		get_formspec = function(self, avail_w)
+			self.resettable = core.settings:has(setting.name)
+			local btn_bind_width = math.max(2.5, avail_w/2)
+			local value = core.settings:get(setting.name)
+			local fs = {
+				("label[0,0.4;%s]"):format(get_label(setting)),
+				("button_key[%f,0;%f,0.8;%s;%s]"):format(
+						btn_bind_width, btn_bind_width-0.8,
+						btn_bind, core.formspec_escape(value)),
+				("image_button[%f,0;0.8,0.8;%s;%s;]"):format(avail_w - 0.8,
+						core.formspec_escape(defaulttexturedir .. "clear.png"),
+						btn_clear),
```

Comment:
Please add a tooltip for this button to clarify what exactly it does.

Reply:
[0119c9f](https://github.com/luanti-org/luanti/pull/15791/commits/0119c9f6e7406652def7e1161e7538219f752a72)